### ISSUE TITLE
`moscot.time.TemporalProblem` as CR kernel

### ISSUE THREAD

**Issue by MUCDK** [theislab/cellrank#982]
As discussed, we should have a CR kernel for `moscot.time.TemporalProblem`. 

@WeilerP let's discuss how to approach this.

**Comment by MUCDK** [comment]
@WeilerP please start with a draft and the implementation on the CR side, then I'll take over from the moscot side.

**Comment by Marius1311** [comment]
Hey! I support this initiative, we really need this. 

I personally prefer the second option - it gives users more control when fitting moscot models. Also, it creates less overhead when we e.g. add or remove parameters on the moscot side, which we would have to mirror in the CR interface then. Also, it would be great if we had one generic interface for all time-structures moscot kernels (as you hinted @MUCDK), including temporal, lineage and spatio-temporal.

**Comment by Marius1311** [comment]
Long term issue: for really large temporal problems, we will have to be careful that we do not blow up memory when we materialize the coupling before sparsifying it by thresholding in CellRank. I just discussed this with @michalk8 - the current implementation of sparsification is nice already because it loads individual problems one by one and sparsifies them sequentially. However, for really large individual problems, even that will be impossible. In these cases, we will have to load chunks of the matrix, sparsify it, and then load the next chunk. 

I don't think we need this in the first iteration implementing this.

**Comment by MUCDK** [comment]
Valid point. I am thinking that this should probably go into moscot.

**Comment by MUCDK** [comment]
I imagine this to have the following structure:

```python
class MoscotKernel(TransportMapKernel2):
    @classmethod
    def load(cls, 
            problem: Type[CompoundBaseProblem],
           time_points: Optional[Tuple[Numeric, Numeric]]=None,
            thresholded: bool = True, self_transitions: Union[
            Literal["uniform", "diagonal", "connectivities", "all"],
            Sequence[Numeric_t],
        ] = SelfTransitions.CONNECTIVITIES,
            conn_weight: Optional[float] = None,
            conn_kwargs: Mapping[str, Any] = MappingProxyType({}),
)-> "TransportMapKernel2":
    self.transport_maps = [p.solution.transport_matrix for p in problem] # if-else statement if thresholded and filtering if time_points!=None
    tmap = self._restich_tmaps(
            self.transport_maps,
            self_transitions=self_transitions,
            conn_weight=conn_weight,
            **conn_kwargs,
        )

      return self

```

**Comment by giovp** [comment]
From a naive point of view, I feel like option 1 could be better simply because cellrank already have a way to handle external kernels and probably a lot of the code for e.g. WOTKernel could be reused (e.g. sparsification of tmap etc.)

I would also be a bit wary about putting the sparsification into moscot cause we don't really have use cases there atm afaik whereas there is clearly the use case for interface to gpcca.

**Comment by MUCDK** [comment]
I disagree, it feels unnecessary to have both ways to compute moscot solutions + maintenance is much worse the first way. 

Plus, we can use the same CellRankkernel for all moscot problems with matrix as output.

**Comment by giovp** [comment]
> I disagree, it feels unnecessary to have both ways to compute moscot solutions + maintenance is much worse the first way.
> 
> Plus, we can use the same CellRankkernel for all moscot problems with matrix as output.

sorry I edited above as I re-read the issue and clarified. I think we want a MoscotKernel in cellrank but we don't want to call the solvers from kernel, so the `load` method is a better option. However, I do maintain that the sparsification should only happen in cellrank for now

**Comment by MUCDK** [comment]
sparsification in CR is complicated as we cannot sparsify the way it is currently done in CR. This is very moscot backend-specific. So we should have it on moscot side IMO

**Comment by giovp** [comment]
> sparsification in CR is complicated as we cannot sparsify the way it is currently done in CR. This is very moscot backend-specific. So we should have it on moscot side IMO

interesting ok, could you elaborate on where are the issue from current approach that moscot should handle?

**Comment by MUCDK** [comment]
sparsification happens [here](https://github.com/theislab/cellrank/blob/525b8477142ded8407734232afa6803d155ed8f5/cellrank/kernels/_transport_map_kernel.py#L119), hence the transport map is instantiated

**Comment by michalk8** [comment]
closed via #1044
